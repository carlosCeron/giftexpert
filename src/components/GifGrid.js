import React from 'react'
import PropTypes from 'prop-types'

const GifGrid = ({category}) => {

  const search = 'naruto';
  const apiKey = 'qNl9pBFbmmnY9rKjK8p30Zh5aCMqJP3C';
  const limit = 10

  const getGifs = async() => {
    const url = `https://api.giphy.com/v1/gifs/search?q=${search}&api_key=${apiKey}&limit=${limit}`;
    const resp = await fetch(url);
    const {data} = await resp.json();
    const gifs = data.map( img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images.downsized_medium.url
      }
    });

    console.log(gifs);
  }

  getGifs();

  return (
    <div>
      <h2>{category}</h2>
    </div>
  )
}

GifGrid.propTypes = {

}

export default GifGrid
