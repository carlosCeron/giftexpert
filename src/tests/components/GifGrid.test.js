import React from 'react';
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

const categoryName = 'Naruto';

describe('Testtin GifGrid Component', () => {


  test('should component has a snapshot', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid  category={categoryName} />);
    expect(wrapper).toMatchSnapshot()
  });


  test('should content after hook was called', () => {
    

    const gifs = [
      {
        id: '1',
        title: 'mi title',
        url: 'http://url.com'
      }
    ]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid  category={categoryName} />);
    expect(wrapper).toMatchSnapshot();
    expect ( wrapper.find('h4').exists()).toBe(false);
    expect ( wrapper.find('GifGridItem').length ).toBe(gifs.length);

  })
  


});