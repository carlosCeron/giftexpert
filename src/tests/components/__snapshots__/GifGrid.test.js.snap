// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testtin GifGrid Component should component has a snapshot 1`] = `
<div>
  <h4>
    Loading...
  </h4>
  <ol
    className="card-grid"
  />
</div>
`;

exports[`Testtin GifGrid Component should content after hook was called 1`] = `
<div>
  <ol
    className="card-grid"
  >
    <GifGridItem
      id="1"
      key="1"
      title="mi title"
      url="http://url.com"
    />
  </ol>
</div>
`;
